2996_reflection/reflection.md

```diff
@@ -2346,30 +2377,30 @@ namespace std::meta {
 ```
 :::
 
-### `name_of`, `display_string_of`, `source_location_of` {#name-loc}
+### `identifier_of`, `display_string_of`, `source_location_of` {#name-loc}
 
 ::: std
 ```c++
 namespace std::meta {
-  consteval auto name_of(info) -> string_view;
-  consteval auto display_string_of(info) -> string_view;
+  consteval auto identifier_of(info) -> string_view;
+  consteval auto u8identifier_of(info) -> u8string_view;
 
-  consteval auto u8name_of(info) -> u8string_view;
+  consteval auto display_string_of(info) -> string_view;
   consteval auto u8display_string_of(info) -> u8string_view;
 
+  consteval auto has_identifier(info) -> bool;
+
   consteval auto source_location_of(info r) -> source_location;
 }
 ```
 :::
 
-If a `string_view` is returned, its contents consist of characters representable by the ordinary string literal encoding only; if any character cannot be represented, it is not a constant expression. If the class of reflected entity cannot possibly have a name, the expression fails to be a constant expression.
+Given a reflection `r` representing a language construct `X` whose declaration introduces an identifier, and if that identifier is representable using the ordinary string literal encoding, then `identifier_of(r)` returns a non-empty `string_view` containing that identifier. Otherwise, it is not a constant expression. Whether a reflected construct has an identifier can be checked with the `has_identifier` metafunction.
 
-Given a reflection `r` that represents a declared entity `X`, `name_of(r)` and `qualified_name_of` return a `string_view` holding the unqualified and qualified name of `X`, respectively. `u8name_of(r)` and `u8qualified_name_of` return the same, respectively, as a `u8string_view`.
-For all other reflections, an empty string view is produced.
-For template instances, the name does not include the template argument list.
+The function `u8identifier_of` returns the same identifier but as a `u8string_view`. Note that since all identifiers can be represented as UTF-8 string literals, `u8identifier_of` never fails to be a constant expression because of representability concerns.
 
-Given a reflection `r`, `display_string_of(r)` and `u8display_string_of(r)` return an unspecified non-empty `string_view` and `u8string_view`, respectively.
-Implementations are encouraged to produce text that is helpful in identifying the reflected construct.
+Given any reflection `r`, `display_string_of(r)` and `u8display_string_of(r)` return an unspecified non-empty `string_view` and `u8string_view`, respectively.
+Implementations are encouraged to produce text that is helpful in identifying the reflected construct (note: as an exercise, the Clang implementation of this proposal implements a pretty-printing `display_string_of` as a non-intrinsic library function).
```

Comment:
Want to add a link to show off?

Reply:
Now that I've gotten rid of the `#define STRING_CONSTANT(str)` macro from the implementation, sure! 🤣